pagopa/io-app
# chore: [IOBP-178,IOBP-184] Native Base dismissal into `IDPayInitiativesListComponents`

## Short description
This PR removes the old `NBListItem` and `NBBadge` into `IDPayInitiativeListComponents` with the new ones available into `@pagopa/io-app-design-system`

## List of changes proposed in this pull request
- Removed `NBList` component into component page;
- Replaced `NBBadge` with the new `Badge` component; 
- Changed Typography references into `PreferencesListItem` component with the `NewH6` (to handle the DS flag) and `LabelSmall`;

## How to test
- Open Wallet section;
- Open a detail card and check and everything should be rendered fine
